File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();

        for(int i=0; i<numRows; i++) {
            List<Integer> ans = new ArrayList<>();
            for(int j=0; j<=i; j++) {
                int sum = 0;
                if(j==i || j==0) {
                    ans.add(1);
                }
                else {
                    ans.add(res.get(i-1).get(j-1) + res.get(i-1).get(j));
                }
            }
            res.add(ans);
        }
        return res;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res = []

        for i in range(numRows) :
            ans = []
            for j in range(i+1):
                if j==0 or j==i:
                    ans.append(1)

                else:
                    sum = res[i-1][j-1] + res[i-1][j]
                    ans.append(sum)

            res.append(ans)
        return res